services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/diagnostics/postTestTable.tsx

```diff
@@ -0,0 +1,194 @@
+import * as React from 'react'
+
+import {
+  correctImage,
+  baseDiagnosticImageSrc,
+} from './shared'
+
+import {
+  helpIcon,
+  Tooltip,
+  smallWhiteCheckIcon,
+} from '../../../../../Shared/index'
+import useWindowSize from '../../../../../Shared/hooks/useWindowSize'
+
+interface StickyTableStyle {
+  position: string,
+  top: number,
+  left: number,
+  right: number,
+  zIndex: number,
+}
+
+const ellipsesIcon = <img alt="Open menu icon" src={`${baseDiagnosticImageSrc}/ellipses_icon.svg`} />
+
+const PostTestAssignmentTable = ({showPostTestAssignmentColumn, students, studentsWhoCompletedDiagnostic, postTestSelections, setPostTestSelections, previouslyAssignedPostTestStudentIds, studentsWhoCompletedAssignedRecommendations, postDiagnosticUnitTemplateId, }) => {
```

Comment:
I think this component duplicates too much code from the older one - I know there's a lot of tightly coupled logic in there, but I think it needs to get DRY-ed up, especially given the complexity of the logic at hand. Having to make any change in two places would get really complicated fast. I think your best bet would be creating a third, more generic component that can just accept props to differentiate itself. The good news is, I think a huge chunk of the code in both places can be deleted - I may be mistaken, but since Jack signed off on no longer using the sticky header, I'm pretty sure everything regarding it can just get deleted, which will really reduce the amount of code that needs deduplication.

Reply:
Note that I think the comments above can all get ignored if I'm right about what can get deleted/de-duplicated.

Reply:
So I ended up deleting this file and simply combining this code into the `RecommendationsTable` component. That component now accepts a boolean indicating whether it's the post test table or not, and then makes some display differentiations based on that. So we longer have that duplicated code and duplicated parameters!